src/parquet/schema/types.h

```diff
@@ -163,15 +163,26 @@ class Node {
     return parent_;
   }
 
+  // ToParquet returns an opaque void* to avoid exporting
+  // parquet::SchemaElement into the public API
+  virtual void ToParquet(void* opaque_element) const = 0;
+
   // Node::Visitor abstract class for walking schemas with the visitor pattern
   class Visitor {
    public:
     virtual ~Visitor() {}
 
     virtual void Visit(const Node* node) = 0;
```

Comment:
I suppose this should be made `Visit(Node* node)` to differentiate with `ConstVisitor`?